dynamic_programming/coin_change.py

```diff
@@ -7,7 +7,7 @@
 """
 
 
-def dp_count(S, m, n):
+def dp_count(S, n):
     """
     >>> dp_count([1, 2, 3], 3, 4)
```

Comment:
```suggestion
from typing import List, Optional


def coin_change(make_change_for: int, coin_types: List[int] = None) -> int:
    """
    >>> coin_change(4, [1, 2, 3])
```
If the doctests are not modified to match the function then our automated tests will fail.

While we are here, let's modify the function to use self-documenting names for function and variables.  Let's also add Python type hints.

Reply:
Before line 24 we could give `coin_types` a default value by adding the line `coin_types = coin_types or (1, 5, 10, 25, 100)`

Reply:
Updated the doctests. 

> Before line 24 we could give coin_types a default value by adding the line coin_types = coin_types or (1, 5, 10, 25, 100)
> This would allow a call such as coin_change(17).

Didn't get this point can you please explain.